torch/_inductor/codegen/cpp.py

```diff
@@ -2036,17 +2036,30 @@ def add_to_dtype(sub_graph: torch.fx.Graph):
                     if src_dtype == torch.bfloat16:
                         # Since we always convert the load/store value to float if the tensor is bfloat16.
                         # Therefore, the reduction should never work with bfloat16 value. Hence, we update
-                        # the bfloat16 reduction by updating the dtype and src_dtype to float.
-                        assert dtype in [torch.float, torch.bfloat16]
-                        _node.args = (
-                            ops,
-                            name,
-                            torch.float,
-                            torch.float,
-                            reduction_type,
-                            index,
-                            value,
-                        )
+                        # the bfloat16 reduction by
+                        #    1) updating the src_dtype to float if dtype is int64
+                        # or 2) updating the dtype and src_dtype to float if else.
+                        assert dtype in [torch.float, torch.bfloat16, torch.int64]
+                        if dtype == torch.int64:
+                            _node.args = (
+                                ops,
+                                name,
+                                torch.int64,
```

Comment:
We could fold the if-condition to here if this is the only difference between the two branches.

Reply:
Thanks. It seems much cleaner.